lib/LedgerSMB/Mailer.pm

```diff
@@ -193,10 +168,9 @@ sub attach {
     # strip path from output name
     my $filename;
     if ($args{filename}) {
-        my $strip = quotemeta $args{strip};
         $filename = $args{filename};
-        $filename =~ s/(.*\/|$strip)//g;
-        }
+        $filename =~ s/(.*\/)//g;
```

Comment:
I suggest to make this more portable (and safer) with File::Basename (core module) and fileparse
http://perldoc.perl.org/File/Basename.html

According to the manual do not use : "basename(1) . It does NOT always return the file name portion of a path as you might expect. To be safe, if you want the file name portion of a path use fileparse() ."

Reply:
I like that suggestion. Are you able to do a PR?

Reply:
I am not able to test it at the moment, can you use something like (replace line 171 and 172 :  
 $filename = fileparse($args{filename},@suffixlist);

I can merge your commit to master, if the test pass.

Reply:
I think actually the filename stripping can and should be removed completely from this method. Mailer should not mangle a parameter that is explicitly given by the caller.